### PR TITLE
Fix/one hot

### PR DESCRIPTION
**Pull request by boczekbartek**

### CODE REVIEW

File: evaluate.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+import torch
+from torch.utils.data import Dataset, DataLoader, RandomSampler, SequentialSampler
+from main import NegCueDataset, BertForNegationCueClassification, MODEL_NAME
+from feature_embeddings import BertPrep
+from tqdm import tqdm
+from sklearn.metrics import classification_report
+
+if __name__ == "__main__":    
+    checkpoint_name = f"{MODEL_NAME}_{999}_{999}"
```
Comment (lagewel001) [inline evaluate.py:11]: 999 was a placeholder name btw :)

File: evaluate.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    
+    lexicals = []
+    # lexicals = ["POS", "Possible_Prefix", "Possible_Suffix"]
+
+    device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
+
+    train_prep = BertPrep("data/SEM-2012-SharedTask-CD-SCO-training-simple-v2-features.csv", lexicals)
+    n_lexicals = 0 if len(lexicals) == 0 else sum(len(next(iter(v.values()))) for v in train_prep.feature_labels.values())
```
Comment (lagewel001) [inline evaluate.py:19]: I suppose it broke on `.next()` when there are no lexicals?

File: evaluate.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    true_tags = []
+    pred_tags = []
+    
+    pred_tags_parsed = []
+    true_tags_parsed = []
+    VERBOSE=False
+
+    tokenizer = train_prep.tokenizer
+    inv_tag_enc = {v:k for k,v in train_prep.tag2idx.items()}
+    with torch.no_grad():
```
Comment (lagewel001) [inline evaluate.py:45]: What is the use of `no_grad()`? I read online that it's common practice when validating, but does the model still update weights even when not calling backwards?

File: main.py [hunk 4]
```diff
@@ -9,45 +9,42 @@
 from transformers.modeling_outputs import TokenClassifierOutput
 from feature_embeddings import BertPrep
 
-EPOCHS = 100
-BATCH_SIZE = 8
+EPOCHS = 25
+BATCH_SIZE = 32
 SEED = 777
 MODEL_NAME = 'neg_cue_detection_model'
```
Comment (lagewel001) [inline main.py:15]: nit

**Review by lagewel001 — APPROVED** [review]
Legend :) Looks very clean

File: evaluate.py (hunk 2 above)
Comment (boczekbartek) [reply evaluate.py:19]: don't remember, or the v has no attribute values()

File: evaluate.py (hunk 3 above)
Comment (boczekbartek) [reply evaluate.py:45]: I am not sure as well, but if optimizer is active it may count this as steps, which can mess it up. (Because for e.g. Adam optimizer keeps gradient from couple of steps before). But it is just my guess..
